includes/wizards/audience/class-audience-content-gates.php

```diff
@@ -434,7 +452,12 @@ public function delete_gate( $request ) {
 		if ( Content_Gate::GATE_CPT !== $gate->post_type ) {
 			return new \WP_Error( 'invalid_gate_type', __( 'Invalid gate type.', 'newspack-plugin' ), [ 'status' => 400 ] );
 		}
-		wp_delete_post( $id, true );
+		$force = $gate->post_status === 'trash';
+		if ( $force ) {
+			wp_delete_post( $id, $force );
+		} else {
+			wp_trash_post( $id );
+		}
```

Comment:
Non-blocking, but that's the default behavior of `wp_delete_post( $id );` without the second argument.